### PR TITLE
feat(onboarding): Add and adapt code from force

### PR DESCRIPTION
**Pull request by damassi**
This adds the [progressive onboarding code from force](https://github.com/artsy/force/blob/main/src/Components/ProgressiveOnboarding), with some small modifications, notably:
- localStorage utils like `get`, etc are no longer exported directly, but can be accessed via the `useLocalStorageUtils` hook
- the context now takes two arguments: `keys: string[]` and `userID?: string` for tracking tips.

### CODE REVIEW

File: src/Context.tsx [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+  timestamp: number
+}
+
+export interface ContextProps {
+  dismissed: DismissedKey[]
+  dismiss: (
+    key: ProgressiveOnboardingKey | readonly ProgressiveOnboardingKey[]
+  ) => void
+  isDismissed: (key: ProgressiveOnboardingKey) => DismissedKeyStatus
+  keys: string[]
```
Comment (dzucconi) [inline src/Context.tsx:30]: IMO this should be a generic

Comment (damassi) [reply src/Context.tsx:30]: Totally, was planning on updating this before merging this PR 👍

Comment (damassi) [reply src/Context.tsx:30]: (gonna come back to this in a follow-up)

**Comment by artsyit** [comment]
:rocket: PR was released in [`v0.2.0`](https://github.com/artsy/dismissible/releases/tag/v0.2.0) :rocket:

**Pull request merged**